### PR TITLE
First PR for upload song page.

### PR DESCRIPTION
**Pull request by Askat08**
This is my first PR. I followed the tasks on [https://trello.com/c/ipDPVDl3/96-story-create-upload-song-component-in-react](url) and created `UploadSong.jsx` and `UploadSong.css`. Then I tried to implement a page with a button and captions using given mockup. In `UploadSong.css` I used `Flexbox` to make new component responsive. 
Hope I didn't do so bad.
I would love to if you could review it and give me a feedback. Thank you.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -10,9 +10,16 @@
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+<<<<<<< HEAD:visual-music-p5-react-wrapper/public/index.html
```
Comment (ArnasDickus) [inline public/index.html]: It looks like you had to fix conflicting files. When fixing it you need to remove these lines of code 
`<<<<<<< HEAD:visual-music-p5-react-wrapper/public/index.html` 
`=======`
`>>>>>>> upstream/development:public/index.html`

File: src/components/SoundPlayer/SoundPlayer.component.jsx [hunk 2]
```diff
@@ -11,8 +11,13 @@
         the part of the song to be played.
 ************************************************************/
 
+<<<<<<< HEAD:visual-music-p5-react-wrapper/src/components/SoundPlayer/SoundPlayer.component.jsx
```
Comment (ArnasDickus) [inline src/components/SoundPlayer/SoundPlayer.component.jsx]: Same thing here.

File: src/components/UploadSong.css [hunk 3]
```diff
@@ -0,0 +1,60 @@
+.boxStyle {
+    display: flex;
+    width: 100vw;
+    align-items: center;
```
Comment (ArnasDickus) [inline src/components/UploadSong.css]: Visual-music uses module.scss not pure css. So rename your file to UploadSong.module.scss and nest elements together no more when 3 levels due to performance reasons

File: src/components/UploadSong.css [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    border: none;
+    outline: none;
+    display: flex;
+    cursor: pointer;
+    margin-bottom: 20px;
+}
+
+.uploadText {
+    display: flex;
+    font-family: Biryani;
```
Comment (ArnasDickus) [inline src/components/UploadSong.css]: font-family is already added in global index.scss. Why did you need to add it here? Also font-style normal is by default.

File: src/components/UploadSong.css [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    font-style: normal;
+    font-weight: 900;
+    font-size: 24px;
+    line-height: 42px;
```
Comment (ArnasDickus) [inline src/components/UploadSong.css]: Try to use rem units instead of px, for accessibility.

File: src/components/UploadSong.css [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    text-align: center;
+
+    color: #4f4e4e;
```
Comment (ArnasDickus) [inline src/components/UploadSong.css]: Repetitive color. Create variable in variable.scss and add it here.

File: src/components/UploadSong.css [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+.fileFormatStyle {
+    display: flex;
+    font-family: Biryani;
+    font-style: normal;
+    font-weight: 900;
+    font-size: 18px;
+    text-align: center;
+
+    color: #4f4e4e;
```
Comment (ArnasDickus) [inline src/components/UploadSong.css]: Repetitive color. Create variable in variable.scss and add it here.

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, thanks for adding PR. So here's my concerns.

1) Did you looked at your file changes, you want to add +2 954 085 lines of code. It seems you included entire node modules folder. I suggest deleting the entire visual-music-p5-react-wrap folder. Also just curious how long did it take you to add upload-song PR to github?

Until you removed node modules folder, I won't be able to review it further.

File: public/index.html (hunk 1 above)
Comment (Askat08) [reply public/index.html]: HI Arnas, I checked the file changes, it turns out I did not carefully check. I'll fix it now and delete entire visual-music-p5-react-wrap folder.
upload-song PR took about 10sec and I did pay attention on it but also same time I did PR after updating my entire file structure, so assumed it maybe it's normal. Now I know, thank you.

File: src/components/SoundPlayer/SoundPlayer.component.jsx (hunk 2 above)
Comment (Askat08) [reply src/components/SoundPlayer/SoundPlayer.component.jsx]: Done.

File: src/components/UploadSong.css (hunk 3 above)
Comment (Askat08) [reply src/components/UploadSong.css]: Done.

**Comment by Askat08** [comment]
I just did another PR with corrected files. Please review it.

**Comment by Askat08** [comment]
First visual-music-p5-react-wrap folder deleted. After i created `UploadSong` folder inside `components` then moved `UploadSong.jsx` and `UploadSong.module.scss` (has renamed from `UploadSong.css`) in to that folder. Also scss variable created for repetitive colors in `variables.scss`. I changed `px` units to `rem` as you suggested and appropriate font size are adjusted. Also I started work on next task in Trello and my `long song titles hidden` commit has added here please review it too.

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, @Askat08 you still have the entire node module folder added. Also there's a conflict with package-lock.json.

1) Delete node modules folder, commit changes.
2) Delete package-lock.json run `npm install`.
3) Adjust .gitignore to prevent node modules files to be uploaded. 
4) Commit changes

File: src/components/PlayerBar/PlayerBar.module.scss [hunk 8]
```diff
@@ -73,13 +98,13 @@
...
         margin: 0 16px;
-        
+
         .progress {
             background-color: rgba(255, 255, 255, 0.8);
             border-radius: inherit;
             position: absolute;
             pointer-events: none;
-            width: 0;
+            width: 99%;
```
Comment (ArnasDickus) [inline src/components/PlayerBar/PlayerBar.module.scss]: Why did you add to 99%? I assume this has to be animated, or we should use some library for that.

File: src/components/PlayerBar/PlayerBar.module.scss [hunk 9]
```diff
@@ -15,11 +15,33 @@
...
     align-items: center;
 }
 
+.volumeButton:hover .volumeSlider {
+    left: 0.6rem;
+    opacity: 1;
+}
+
+.volumeSlider {
+    left: 0rem;
```
Comment (ArnasDickus) [inline src/components/PlayerBar/PlayerBar.module.scss]: This would be nitpicking, but you don't need to add rem to 0. 0 = 0 no matter if it's px or rem.

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, so here's my review:
1) The player bar sound seems to be operational. I don't think we can do much about visuals, changing when sound is decreasing.
2) You seem to have added 2 different trello tasks in the same PR. I suggest next time creating 2 different PR for that.
[upload-song](https://trello.com/c/ipDPVDl3/96-story-create-upload-song-component-in-react)
[player-bar](https://trello.com/c/KDIpnjgf/104-improve-player-bar)

3) There is in App_visualmusic padding-bottom 64px. That you didn't added, but it leaves white space, so I suggest removing it. 
remove-padding-bottom

4) Upload new song not mobile responsive. You might need, to decrease content amount in order for it to fit.
upload-new-song not mobile responsive

You seem to have taken 2 tasks on one PR and getting stuck on player bar functionality. My suggestion is this: Fix current issues, I mentioned, after you fix it I would approve this PR and would create additional tasks, to improve the player bar, upload song component. Or if you want you can continue implementing player bar functionalities. But I would like to approve this PR by the end of the week.

File: src/components/PlayerBar/PlayerBar.module.scss (hunk 8 above)
Comment (Askat08) [reply src/components/PlayerBar/PlayerBar.module.scss]: sorry, i was changing `width` percentage just to get idea of how should look like animated version. Should have change back to `0`.

File: src/components/PlayerBar/PlayerBar.module.scss (hunk 9 above)
Comment (Askat08) [reply src/components/PlayerBar/PlayerBar.module.scss]: > This would be nitpicking 

Not at all, this is actually something I should know already. Will fix it now. Thank you.

**Review by ArnasDickus — APPROVED** [review]
The upload song is responsive. I will create smaller tasks to fully finish the player bar. As well as connect UploadSong with playerBar.

**Comment by Askat08** [comment]
Whooh finally we did it. :) Thank you @ArnasDickus
